bot/cogs/allow_deny_lists.py

```diff
@@ -95,9 +95,21 @@ async def _delete_data(self, ctx: Context, allowed: bool, list_type: ValidAllowD
         """Remove an item from an allow or denylist."""
         item = None
         allow_type = "whitelist" if allowed else "blacklist"
+        id_converter = IDConverter()
 
-        log.trace(f"Trying to delete the {content} item from the {list_type} {allow_type}")
+        # If this is a server invite, we need to convert it.
+        if list_type == "GUILD_INVITE" and not id_converter._get_id_match(content):
+            log.trace(f"{content} is a guild invite, attempting to validate.")
+            validator = ValidDiscordServerInvite()
+            guild_data = await validator.convert(ctx, content)
+
+            # If we make it this far without raising a BadArgument, the invite is
+            # valid. Let's convert the content to an ID.
+            log.trace(f"{content} validated as server invite. Converting to ID.")
+            content = guild_data.get("id")
```

Comment:
This is pretty similar to the `_add_data` code. Does it make sense to refactor it into a separate function?

Reply:
Yep! fixed in a7a3e29ca901b84570e5a1ff1e4c2bcf22b86552